### PR TITLE
[OSCI] deleting folders docs/, docs-site/ and examples/ in packages/react-da…

### PR DESCRIPTION
**Pull request by BigSamu**
…tepicker

### Description
Cleaning up of `react-datepicker` package 

### Issues Resolved

- Fixes Issue #766
- Deleting folders docs/, docs-site/ and examples/ 
- File highlight.js not found inside the directory `packages/react-datepicker`

### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,7 +30,8 @@
 ### 🛠 Maintenance
 
 - Update caniuse database ([#1046](https://github.com/opensearch-project/oui/pull/1046))
-- Bump TypeScript to v4.6.4 ([#879](https://github.com/opensearch-project/oui/pull/879))
+- Bump TypeScript to v4.6.4 ([#879](https://github.com/opensearch-project/oui/pull/879)
```
Comment (BSFishy) [inline CHANGELOG.md:33]:
```suggestion
- Bump TypeScript to v4.6.4 ([#879](https://github.com/opensearch-project/oui/pull/879))
```

Looks like a `)` got cut off here accidentally

**Review by BSFishy — COMMENTED** [review]
Looks good, other than I'd like to have `highlight.js` to be removed from deps of this package

**Comment by BigSamu** [comment]
> Looks good, other than I'd like to have `highlight.js` to be removed from deps of this package

@BSFishy `highlight.js` has been removed.

File: CHANGELOG.md (hunk 1 above)
Comment (BigSamu) [reply CHANGELOG.md]: @BSFishy fixed the error.

**Review by BSFishy — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 2]
```diff
@@ -31,6 +31,7 @@
 
 - Update caniuse database ([#1046](https://github.com/opensearch-project/oui/pull/1046))
 - Bump TypeScript to v4.6.4 ([#879](https://github.com/opensearch-project/oui/pull/879))
+- Cleaning up of react-datepicker([#1067](https://github.com/opensearch-project/oui/pull/1067))
```
Comment (JohnathonBowers) [inline CHANGELOG.md:34]: To keep the language consistent with other change log messages, I would suggest saying "Clean up react-datepicker".

Comment (joshuarrrr) [reply CHANGELOG.md:34]:
```suggestion
- Clean up `react-datepicker` package to remove unnecessary directories and files ([#1067](https://github.com/opensearch-project/oui/pull/1067))
```

Comment (BigSamu) [reply CHANGELOG.md]: The last suggestion from @joshuarrrr has been incorporated in CHANGELOG.md

**Review by JohnathonBowers — APPROVED** [review]
Looks good to me! Good work, Samuel!

**Review by joshuarrrr — APPROVED** [review]

**Pull request merged**